### PR TITLE
Add validation and unit test for empty credentials

### PR DESCRIPTION
**Pull request by intojhanurag**
This PR improves the NewPromptForCredentials function by adding validation to handle empty username or password inputs. It ensures users cannot proceed with blank credentials.

Additionally, a new unit test has been added to verify that the function correctly returns an error when either field is left empty. 

Screenshot 2025-10-30 000714

### CODE REVIEW

**Comment by intojhanurag** [comment]
Hey @gauron99 
I’m just curious, I only added one extra test case and a small validation check for empty credentials, but it seems a lot of E2E and integration tests started failing.
My local make test passed successfully.
Would love to understand this behavior better.

**Comment by gauron99** [comment]
helm installation is failing, unrelated; its failing everywhere

**Comment by intojhanurag** [comment]
Haha, thank God,  I thought I did something blunder 😅

**Comment by gauron99** [comment]
no worries, thats why we have tests to catch any possible "blunders" anyways... this case is not that though. its azure front door. We will rerun tests when the service seems up and running again

**Comment by intojhanurag** [comment]
Hey @gauron99 , Is it good to go now ?

**Comment by gauron99** [comment]
you have some third party changes

File: cmd/prompt/prompt_test.go [hunk 1]
```diff
@@ -79,4 +79,16 @@ func Test_NewPromptForCredentials(t *testing.T) {
 			}
 		})
 	}
+	t.Run("empty username or password returns error", func(t *testing.T) {
```
Comment (gauron99) [inline cmd/prompt/prompt_test.go]: why not put this as part of the for loop above?

**Comment by intojhanurag** [comment]
> you have some third party changes

Yeah , Due to "code styles" failure I ran the goimport -w . command in the repo .

File: cmd/prompt/prompt_test.go (hunk 1 above)
Comment (intojhanurag) [reply cmd/prompt/prompt_test.go]: Hey @gauron99 , I kept the "empty username/password" case outside because it tests a different outcome (an error condition).
